### PR TITLE
feat: multi-query support for reports

### PR DESCRIPTION
**Pull request by tomwwinter**

### CODE REVIEW

File: docs/api-specs/reporting-api-v1.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ info:
   description: |-
     API to manage reports that provide data calculated based on any entities of the Aam Digital system
     and offer notifications when data of such reports changes.
-  version: 1.0.0-alpha.4
+  version: 1.0.0-alpha.5
```
Comment (sleidig) [inline docs/api-specs/reporting-api-v1.yaml:7]: maybe we should have an official version 1.0 by now (it is used in production, after all) and have these changes as a 1.1-beta ?

**Review by sleidig — COMMENTED** [review]
Code looks good to me, as far as I can tell.
I like some of the re-structuring and the placeholder and transformation design!

This contains a lot of refactoring again (which is great! But as we really need to ensure reliable functioning nowadays, separating refactoring and functional changes in separate PRs wherever possible would help with testing and review).

File: docs/api-specs/reporting-api-v1.yaml
Comment (sleidig) [inline docs/api-specs/reporting-api-v1.yaml:1]: It seems like we do not specify the actual ReportData structure here yet (which has been changed / added for v2 now also). I think this is something API clients need to know and should be added here to the ReportData schema

File: application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/report/storage/ReportConfigV1Entity.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Replaced by: ReportConfigEntity
+ */
+data class ReportConfigV1Entity(
+    @JsonProperty("_id")
+    val id: String,
+    @JsonProperty("_rev")
+    val rev: String,
+    val title: String,
+    val version: Int = 1,
```
[the remark this answers is not in the document]
Comment (tomwwinter) [reply application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/report/storage/ReportConfigV1Entity.kt:17]: Yes, if no version is set in the doc, the default value is applied.

File: application/aam-backend-service/src/test/resources/database/documents/ReportCalculation:2.json [hunk 3]
```diff
@@ -3,10 +3,7 @@
   "report": {
     "id": "ReportConfig:1"
   },
-  "args": {
-    "from": "2024-01-01T00:00:00Z",
-    "to": "2024-04-30T00:00:00Z"
-  },
+  "args": {},
```
[the remark this answers is not in the document]
Comment (tomwwinter) [reply application/aam-backend-service/src/test/resources/database/documents/ReportCalculation:2.json:6]: Yes, this test was only not correct. There were args passed, but the query does not expect any args.

File: application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/error/AamException.kt [hunk 4]
```diff
@@ -14,12 +14,14 @@ class InternalServerException(
     code: AamErrorCode,
 ) : AamException(message, cause, code)
 
+// todo merge with network exception and rename to IOException
```
[the remark this answers is not in the document]
Comment (tomwwinter) [reply application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/error/AamException.kt]: something for later, i created an issue for that.

File: application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/report/core/SimpleReportSchemaGenerator.kt [hunk 5]
```diff
@@ -23,8 +24,10 @@ class SimpleReportSchemaGenerator : ReportSchemaGenerator {
 
     override fun getAffectedEntities(report: Report): List<String> {
         val sqlFromTableRegex: Pattern = Pattern.compile("FROM\\s+(\\w+)")
+        val query: ReportItem.ReportQuery =
+            report.items.first { it is ReportItem.ReportQuery } as ReportItem.ReportQuery
         return sqlFromTableRegex
-            .matcher(report.query)
+            .matcher(query.sql) // todo analyse all queries here
```
[the remark this answers is not in the document]
Comment (tomwwinter) [reply application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/report/core/SimpleReportSchemaGenerator.kt:13]: Yes, thanks. I added the implementation and tests.

File: application/aam-backend-service/src/test/resources/database/documents/ReportCalculation:2.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sleidig) [reply application/aam-backend-service/src/test/resources/database/documents/ReportCalculation:2.json:6]: do we have any tests with time period args in place? Then I guess we are fine

File: docs/api-specs/reporting-api-v1.yaml
Comment (sleidig) [reply docs/api-specs/reporting-api-v1.yaml:1]: yeah, can we maybe add 2 examples and link to the interfaces of the code that define the response object type?

**Pull request merged**